facebook/jest
# Support for --preserve-symlinks

## Summary

This is my attempt to fix https://github.com/facebook/jest/issues/5356 - I have been making very slow progress over the past months on this so I thought I would just open this PR to increase visibility and maybe someone else is inclined to help finish this up quicker.

For motivation see the discussion at #5356 

Whit is still missing:
 - Making sure the flag is piped through correctly everywhere, it is a bit difficult to keep track of all the places, especially given the time frame I have worked on this but as far as I remember only packages/jest-runtime/src/script_transformer.js is missing and I think making sure to whoever is calling the jest-resolve package to ensure the defaultResolver gets the flag set correctly
 - changelog entries
 - tests



## Test plan

Still need to add tests but the test plan would be the follwoing:

1. Create a test file, e.g. sum.test.js
2. Create a symlink of that file in a different directory (e.g. `ln -s ../sum.test.js ./sum.test.js`)
3. Make sure you are in that directory that contains the symlink and run `jest --preserve-symlinks sum.test.js
4. See that the test is actually running
